internal/state/dataplane/configuration.go

```diff
@@ -24,8 +25,7 @@ type Configuration struct {
 	// Upstreams holds all unique Upstreams.
 	Upstreams []Upstream
 	// BackendGroups holds all unique BackendGroups.
-	// FIXME(pleshakov): Ensure Configuration doesn't include types from the graph package.
-	BackendGroups []graph.BackendGroup
+	BackendGroups []BackendGroup
```

Comment:
there is a comment in the line `181` (`// We are only interested in BackendGroups that have at least one Backends.`) that explains that this field doesn't hold all BackendGroups. Should we reflect that in the name of the field or perhaps the comment for the field ?

Reply:
Good call out. Instead of renaming, I can actually remove the check on line 182. The `nginx/config` package filters out BackendGroups that don't need splits.